src/main.rs

```diff
@@ -424,6 +425,36 @@ pub enum ContextPage {
     Settings,
 }
 
+#[derive(Clone, Debug, Default)]
+pub struct DocumentStatistics {
+    character_count: usize,
+    character_count_no_spaces: usize,
+    line_count: usize,
+    word_count: usize,
+}
+
+fn calculate_document_statistics(lines: Vec<String>) -> DocumentStatistics {
```

Comment:
This should be borrowed since it doesn't need ownership, and that sounds like a very costly vec.